src/matrix_validator/validator_polars.py

```diff
@@ -22,152 +36,140 @@ def __init__(self):
         """Create a new instance of the polars-based validator."""
         super().__init__()
 
-    def validate(self, nodes_file_path, edges_file_file_path):
+    def validate(self, nodes_file_path, edges_file_path):
         """Validate a knowledge graph as nodes and edges KGX TSV files."""
+        validation_reports = []
+
+        bm_prefixes = self.get_biolink_model_prefix_keys()
+        bm_knowledge_level_keys = self.get_biolink_model_knowledge_level_keys()
+        bm_agent_type_keys = self.get_biolink_model_agent_type_keys()
+
         if nodes_file_path:
-            validate_kg_nodes(nodes_file_path, self.output_format, self.get_report_file())
+            validation_reports.extend(validate_kg_nodes(bm_prefixes, nodes_file_path))
 
-        if edges_file_file_path:
-            validate_kg_edges(edges_file_file_path, self.output_format, self.get_report_file())
+        if edges_file_path:
+            validation_reports.extend(validate_kg_edges(bm_prefixes, bm_knowledge_level_keys, bm_agent_type_keys, edges_file_path))
 
+        if nodes_file_path and edges_file_path:
+            validation_reports.extend(validate_node_ids_in_edges(nodes_file_path, edges_file_path))
+
+        # Write validation report
+        self.write_report(validation_reports)
+        logging.info(f"Validation report written to {self.get_report_file()}")
+
+
+def validate_node_ids_in_edges(nodes, edges):
```

Comment:
why is this function not in a separate file like src/matrix_validator/checks/check_edge_ids_in_node_ids.py?

Reply:
It is not a check which lists offending nodes, rather a check that counts offending nodes.

Reply:
How about renaming the function to 'validate_nodes_and_edges()'?

Reply:
I renamed the function.